ocean-components/src/main/java/br/com/useblu/oceands/ui/compose/stringmask/CurrencyMaskVisualTransformation.kt

```diff
@@ -0,0 +1,54 @@
+package br.com.useblu.oceands.ui.compose.stringmask
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import kotlin.math.absoluteValue
+
+class CurrencyMaskVisualTransformation : VisualTransformation {
+    override fun filter(text: AnnotatedString): TransformedText {
+        var outputText = ""
+
+        if (text.length < 3) {
+            return TransformedText(text, OffsetMapping.Identity)
+        }
+
+        var charCount = -1
```

Comment:
Por que -1?

Reply:
vou tentar melhorar esse codigo, nao ficou mt claro o que ele faz,
mas é pra quando chegar no terceiro digito ele add o ".", fiz o contador começar em -1 e a condicional ser true quando o resto da divisao por 3 for igual a 2